### ISSUE TITLE
Uninstalling causes loss of DNS (Only when installing unbound)

### ISSUE THREAD

**Issue by zfz7** [complexorganizations/wireguard-manager#7]
**Describe the bug**
After uninstalling using the script dns requests will not resolve even after a reboot. 

**To Reproduce**
1. Installing using all default settings (make sure to install unbound)
2. `ping www.google.com` see no issues
3. Uninstall using the script
4. `ping www.google.com` -> `Temporary failure in name resolution`
5. Reboot
6. `ping www.google.com` -> `Temporary failure in name resolution`

**Expected behavior**
I am hoping that an uninstall will not cause a dns failure 

**Screenshots**
NA

**Desktop (please complete the following information):**
 - OS: Ubuntu 20.04
 - Browser: NA
 - Cloud Provider: AWS

**Smartphone (please complete the following information):**
 - Device: NA
 - OS:NA
 - Browser: NA
 - Version:NA

**Additional context**
If you do not install unbound there doesn't seem to be an issue

**Comment by Prajwal-Koirala** [comment]
https://github.com/complexorganizations/wireguard-manager/pull/8

**Comment by zfz7** [comment]
Wow you're super fast! This [commit](https://github.com/complexorganizations/wireguard-manager/pull/8/commits/b493a9ca9e54431816f25e93ac12f980ce9f3a9c) did not work but just saw you added another one. Let me check that. thanks again!

**Comment by zfz7** [comment]
Hmm didnt work either same error...

**Comment by Prajwal-Koirala** [comment]
what os are u using?
i tested it on ubuntu 20 and seems to work fine.

**Comment by zfz7** [comment]
Ubuntu 20.04 on AWS. Hmm let me try again

**Comment by Prajwal-Koirala** [comment]
testing it again.

**Comment by zfz7** [comment]
Didn't work :( Here is my /etc/resolv.con file after the uninstall
```
# Dynamic resolv.conf(5) file for glibc resolver(3) generated by resolvconf(8)
#     DO NOT EDIT THIS FILE BY HAND -- YOUR CHANGES WILL BE OVERWRITTEN
# 127.0.0.53 is the systemd-resolved stub resolver.
# run "systemd-resolve --status" to see details about the actual nameservers.

# 
search ec2.internal
options edns0
```

**Comment by Prajwal-Koirala** [comment]
i see the issue, give me 2 mins

**Comment by Prajwal-Koirala** [comment]
This should fix it,

I just tested it and works perfectly fine.

https://github.com/complexorganizations/wireguard-manager/commit/336ae60e1d1c4c7609dae8e252dad173f669f666

**Comment by zfz7** [comment]
Man I hate to say this but it didn't work for me. My steps are 

1. new aws instance Ubuntu Server 20.04 LTS (HVM), SSD Volume Type - ami-068663a3c619dd892 (64-bit x86)

- I am run all default VPS settings, no special rules 

1. `sudo apt-get update && sudo apt-get upgrade`

1. `sudo apt-get install jq`

1. Then pull your script and run it..

**Comment by zfz7** [comment]
Error seems to happen on 18.04 as well.

**Comment by zfz7** [comment]
I think I know what is happening, there is an internal error when you call `chattr -i`

```
ubuntu@ip-172-31-66-xxx:~$ sudo chattr -i /etc/resolv.conf
sudo: unable to resolve host ip-172-31-66-xxx: Resource temporarily unavailable
```
to fix it I need to add `172.31.66.xxx ip-172-31-66-xxx` to `/etc/hosts`

**Comment by Prajwal-Koirala** [comment]
added a `chattr` requirement, also the issue shouldn't be with the nameservers 

also i am going to test this on AWS ec2 now.

**Comment by zfz7** [comment]
Yeah it seems like it is a aws issue see [this](https://stackoverflow.com/questions/33441873/aws-error-sudo-unable-to-resolve-host-ip-10-0-xx-xx/33443018). To fix it you may need to run something like this `sudo sed -i /etc/hosts -e "s/^127.0.0.1 localhost$/127.0.0.1 localhost $(hostname)/"`

**Comment by Prajwal-Koirala** [comment]
# Fixed

### Before Install
```
nameserver 8.8.8.8
search ec2.internal
options edns0
```
### After Install
```
#nameserver 8.8.8.8
#search ec2.internal
options edns0
nameserver 127.0.0.1
nameserver ::1
```
### After Uninstall
```
nameserver 8.8.8.8
search ec2.internal
options edns0
```

**Comment by zfz7** [comment]
Hey that code unfortunately still does not work. Here is the output running that code. 
**Before Install**
```
nameserver 127.0.0.53
options edns0
search ec2.internal
```
**After Install**
```
#nameserver 127.0.0.1
#search ec2.internal
options edns0
nameserver 127.0.0.1
nameserver ::1
```
**After Unintstall**
```
search ec2.internal
options edns0
```

I did a little digging I think Ubuntu handles /etc/resolv.conf a little differently and you shouldn't touch it. My issue can be solved by [this](https://github.com/complexorganizations/wireguard-manager/pull/9) PR

**Comment by Prajwal-Koirala** [comment]
Please update to the latest version of the script, it should be fixed now.

**Comment by zfz7** [comment]
No didn't work. I think if you touch resolv.conf the uninstall process will not restore it.

**Comment by Prajwal-Koirala** [comment]
try this final one,
if this doesn't work something is wrong with ur server.

https://github.com/complexorganizations/wireguard-manager/pull/10

**Comment by zfz7** [comment]
That works! 

Before Install
```
# This file is managed by man:systemd-resolved(8). Do not edit.
#
# This is a dynamic resolv.conf file for connecting local clients to the
# internal DNS stub resolver of systemd-resolved. This file lists all
# configured search domains.
#
# Run "resolvectl status" to see details about the uplink DNS servers
# currently in use.
#
# Third party programs must not access this file directly, but only through the
# symlink at /etc/resolv.conf. To manage man:resolv.conf(5) in a different way,
# replace this symlink by a static file or a different symlink.
#
# See man:systemd-resolved.service(8) for details about the supported modes of
# operation for /etc/resolv.conf.

nameserver 127.0.0.53
options edns0
search ec2.internal
```
After Install
```
nameserver 127.0.0.1
nameserver ::1
```
After Uninstall
```                           
# Dynamic resolv.conf(5) file for glibc resolver(3) generated by resolvconf(8)
#     DO NOT EDIT THIS FILE BY HAND -- YOUR CHANGES WILL BE OVERWRITTEN
# 127.0.0.53 is the systemd-resolved stub resolver.
# run "systemd-resolve --status" to see details about the actual nameservers.

nameserver 127.0.0.53
search ec2.internal
options edns0
```
